src/Network.cpp

```diff
@@ -860,9 +861,40 @@ Network::Netresult Network::get_scored_moves(
     if (ensemble == DIRECT) {
         assert(rotation >= 0 && rotation <= 7);
         result = get_scored_moves_internal(state, planes, rotation);
+    } else if (ensemble == MULTI_AVG && rotation > 1) {
+        // MULTI_AVG rotation is the # of random rotations to provide
+        assert(rotation >= 1 && rotation <= 8);
+        std::vector<int> r_list{0, 1, 2, 3, 4, 5, 6, 7};
+        // Need to account for extra 2 in m_squaresize.
+        float policy[(BOARD_SIZE + 2) * (BOARD_SIZE + 1)] = {0};
+
+        if (rotation < 8) {
+            std::random_device rd;
+            std::mt19937 g(rd());
```

Comment:
This is completely unneeded, we already do a lot of work to correctly initialize a suitable RNG. IIRC you can see how to use it with shuffle in Training.cpp.